include/cuda/std/detail/__atomic

```diff
@@ -195,6 +192,126 @@ __host__ __device__ auto __cxx_atomic_alignment_unwrap(_Tp __value)
     return __cxx_atomic_alignment_unwrap(__value, integral_constant<bool, _LIBCUDACXX_ALIGNOF(_Tp) == sizeof(_Tp)>{});
 }
 
+template <typename _Tp>
+__host__ __device__ constexpr ptrdiff_t __cxx_atomic_skip(ptrdiff_t __value) {
+    using __t = decltype(*__cxx_atomic_alignment_unwrap(_Tp{}, integral_constant<bool, _LIBCUDACXX_ALIGNOF(_Tp) == sizeof(_Tp)>{}));
```

Comment:
This invokes unwrap to get the underlying type (if needed) for incrementing pointers

eg: 
 * `int* == 4*increment`
 * `long* == 8*increment`
 * `custom_class* == sizeof(custom_class)*increment`

Reply:
I wonder if instead we should just expose a `value_type` in the alignment wrapper and query that here.